macaroonbakery/checkers/utils.py

```diff
@@ -7,7 +7,7 @@ def condition_with_prefix(prefix, condition):
 
     If the prefix is non-empty, a colon is used to separate them.
     '''
-    if prefix == '':
+    if prefix == '' or prefix is None:
```

Comment:
you can replace this conditional with the more python version

if not prefix:

Reply:
But that wouldn't have quite the same semantics  - it would trigger if prefix was (for example) the number 0 or an empty list, which wouldn't be quite right (currently the function will raise an exception if you do that). I think I'd prefer to leave it explicit.